be/src/formats/parquet/meta_helper.h

```diff
@@ -0,0 +1,132 @@
+            return nullptr;
+        }
+
+        /**
+        * Metadata for each column chunk in row group must have the same order as the SchemaElement list in FileMetaData.
+        **/
+        return &row_group.columns[it->second].meta_data;
+    }
+
+    const ParquetField* get_parquet_field(const std::unordered_map<std::string, size_t>& column_name_2_pos_in_meta,
+                                          const std::string& col_name) const {
+        auto it = column_name_2_pos_in_meta.find(col_name);
+        if (it == column_name_2_pos_in_meta.end()) {
+            return nullptr;
+        }
+        return _file_metadata->schema().get_stored_column_by_idx(it->second);
+    }
+
+protected:
+    GroupReaderParam::Column _build_column(int col_idx_in_parquet, int col_idx_in_chunk,
+                                           const tparquet::Type::type& col_type_in_parquet,
+                                           const TypeDescriptor& col_type_in_chunk, const SlotId& slot_id,
+                                           const TIcebergSchemaField* t_iceberg_schema_field = nullptr) const {
+        GroupReaderParam::Column column{};
+        column.col_idx_in_parquet = col_idx_in_parquet;
+        column.col_idx_in_chunk = col_idx_in_chunk;
+        column.col_type_in_parquet = col_type_in_parquet;
+        column.col_type_in_chunk = col_type_in_chunk;
+        column.slot_id = slot_id;
+        column.t_iceberg_schema_field = t_iceberg_schema_field;
+        return column;
+    }
+
+    bool _case_sensitive = false;
+    std::shared_ptr<FileMetaData> _file_metadata;
+};
+
+class ParquetMetaHelper : public MetaHelper {
+public:
+    ParquetMetaHelper(std::shared_ptr<FileMetaData> file_metadata, bool case_sensitive) {
+        _file_metadata = file_metadata;
+        _case_sensitive = case_sensitive;
+    }
+    ~ParquetMetaHelper() override = default;
+
+    void set_existed_column_names(std::unordered_set<std::string>* names) const override;
+    void build_column_name_2_pos_in_meta(std::unordered_map<std::string, size_t>& column_name_2_pos_in_meta,
+                                         const tparquet::RowGroup& row_group,
+                                         const std::vector<SlotDescriptor*>& slots) const override;
+    const void prepare_read_columns(const std::vector<HdfsScannerContext::ColumnInfo>& materialized_columns,
+                                    std::vector<GroupReaderParam::Column>& read_cols,
+                                    bool& _is_only_partition_scan) const override;
+
+private:
+};
+
+class IcebergMetaHelper : public MetaHelper {
+public:
+    IcebergMetaHelper(std::shared_ptr<FileMetaData> file_metadata, bool case_sensitive,
+                      const TIcebergSchema* t_iceberg_schema) {
+        _file_metadata = file_metadata;
+        _case_sensitive = case_sensitive;
+        _t_iceberg_schema = t_iceberg_schema;
+        DCHECK(_t_iceberg_schema != nullptr);
+    }
```

Comment:
what about we update be first and then fe? we'd better give some prompt information here.

Reply:
if it go non-iceberg-schema branch, schema evolution not work, where can we give some message?

Reply:
Just like before, query will return empty result.

Reply:
ok